lib/plugins/features/movies/controller.js

```diff
@@ -30,3 +31,25 @@ exports.findAll = (params) => {
   }).fetchAll();
 
 }
+
+exports.assignLocation = (movie_id, location_id) => {
+  let movie;
+  return new Movie({ id: movie_id }).fetch({
+    withRelated: ['locations']
+  })
+  .then((m) => {
+    movie = m;
+    return new Location({ id: location_id }).fetch();
+  })
+  .then((location) => {
+    const associatedLocations = movie.related('locations').models.map((model) => model.id);
+    if (!(associatedLocations.indexOf(location_id) >= 0)) {
```

Comment:
It'd be awesome if we didn't have to do this on the application level, and instead did it with a database query at the beginning. Something like:
1. See if the `movie` `location` pair exists
2. Create it
3. Fetch Movie with related locations

That said, I'm not seeing how to do that in Bookshelf without simply digging into the Knex itself. So I'd say nothing to fix here!